subprojects/stdlib/stdlib.c

```diff
@@ -0,0 +1,32 @@
+#include "library_api.h"
+#include <stdio.h>
+
+const char* day_of_week(char* arg) {
+    const char* days_of_week[] = {
+        "Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"
+    };
+
+    int year, month, day;
+
+    if (sscanf(arg, "%d-%d-%d", &year, &month, &day) != 3) {
+        return "[ERROR: Invalid date format]";
+    }
+
+    // Zeller
+    if (month < 3) {
+        month += 12;
+        year -= 1;
+    }
+
+    int k = year % 100;
+    int j = year / 100;
+    int h = (day + (13 * (month + 1)) / 5 + k + k / 4 + j / 4 - 2 * j) % 7;
+
+    int day_of_week = (h + 5) % 7;
+
+    return days_of_week[day_of_week];
+}
+
+DEFINE_LIBRARY(
+    FUNCTION("day_of_week", day_of_week)
```

Comment:
I still don't think this macro is necessary, it only declares a struct and does a casting.

However, the casting of `day_of_week` is needed here because the return type is different from `function pointer`. One returns `char *` and the other returns `const char*`.

If you fix this inconsistency, the casting is not necessary, and the removal of the macro makes sense.